src/controllers/FacetQueryController.ts

```diff
@@ -379,7 +379,12 @@ export class FacetQueryController {
     // Replace the wildcard (*) for a regex match (.*)
     // Also replace the (?) with "any character once" since it is also supported by the index
     return _.some(this.facet.options.allowedValues, allowedValue => {
-      const regex = new RegExp(`^${allowedValue.replace(/\*/g, '.*').replace(/\?/g, '.')}$`, 'gi');
+      const regexContent = allowedValue
+        .replace(/[.+^${}()|[\]\\]/g, '\\$&')
+        .replace(/\*/g, '.*')
+        .replace(/\?/g, '.');
```

Comment:
Would it be a good idea to split this across multiple variables and/or multiple functions for readability?

Reply:
you don't think having this on multiple lines is readable enough?